### PR TITLE
Integration test feature

### PR DESCRIPTION
**Pull request by Alusp**
## I implemented the following in this PR:

- [ ] Solve N+1 Problem
- [ ] Integration specs
- [ ] Use Capybara to write integration tests for each view in your project.
- [ ] User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.

- [ ] user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.

- [ ] User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.

- [ ] Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/users/users_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      expect(page).to have_content(@post3.title)
+    end
+
+    it 'displays button to see all posts' do
+      expect(page).to have_link('Show All Posts')
+    end
+
+    it 'click a user post and redirects to post show page' do
+      click_link 'Show All Posts'
+      # expect(current_path).to eq(post_path(@user1))
```
Comment (BertrandConxy) [inline spec/views/users/users_show_spec.rb]: - [ ] Please remove the above commented line of code to keep your test file codebase clean 😃 👍

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @Alusp, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 

## Highlights 💪 
- No linter errors
- Tests passes 💪 
- correct flow is used
- good PR

## Required Changes ♻️

- [ ] Great job! In addition to what you have done, I would suggest that you solve n+1 problem in the `posts#index` action where I suggest that you use `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts...` Take a look at the screenshot below 👇 
image

_Check the comments under the review._

## Optional suggestions

- [ ] [OPTIONAL] Please rename the tests folder from `views` to `features`  where you put tests concerning pages. This is a convention and It is better to follow the convention

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alusp** [comment]
Alright thanks very much for the review @BertrandConxy

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Alusp👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alusp** [comment]
Thanks for the review @thecodechaser

**Pull request merged**